### PR TITLE
Use a generic markup class to display externally rendered files and diffs

### PR DESCRIPTION
**Pull request by 6543**
This pull is based on #12261

---

This PR will be a follow-on from https://github.com/go-gitea/gitea/issues/8299 and https://github.com/go-gitea/gitea/issues/7614.

It builds on the work done in https://github.com/go-gitea/gitea/pull/8300 and the patch https://github.com/go-gitea/gitea/pull/8357

The idea is to have a generic class `.markup` that has some default rendering, that is near identical to the current `.markdown`

<details><summary>More details</summary>

The current `.markdown` is defined at 

https://github.com/go-gitea/gitea/blob/master/web_src/less/_markdown.less

and overwritten at:

https://github.com/go-gitea/gitea/blob/84a419de98701fe1e8a0d7f1be5d82c7f5e4ac92/web_src/less/themes/theme-arc-green.less#L851-L885

and

https://github.com/go-gitea/gitea/blob/84a419de98701fe1e8a0d7f1be5d82c7f5e4ac92/web_src/less/themes/theme-arc-green.less#L1856-L1867

</details>

## :warning: BREAKING :warning: 

Administrators with css/less customization targeting the  `.markdown` class should update these to use `.markup`.

### CODE REVIEW

File: docs/content/doc/advanced/external-renderers.en-us.md [hunk 1]
```diff
@@ -98,3 +98,71 @@ Once your configuration changes have been made, restart Gitea to have changes ta
 
 **Note**: Prior to Gitea 1.12 there was a single `markup.sanitiser` section with keys that were redefined for multiple rules, however,
 there were significant problems with this method of configuration necessitating configuration through multiple sections.
+
+## Customising CSS
```
Comment (silverwind) [inline docs/content/doc/advanced/external-renderers.en-us.md]:
```suggestion
## Customizing CSS
```

American variant is probably more proper for docs.

File: web_src/less/_markup.less [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.markdown:not(code) {
+.markup:not(code) {
```
Comment (silverwind) [inline web_src/less/_markup.less]: Maybe drop this `:not(code)` and see what breaks, if anything at all? Might need to check git history why it was added in first place.

Comment (silverwind) [reply web_src/less/_markup.less]:
```
rg markdown.+code templates
rg code.+markdown templates
```
No relevant matches.

Comment (silverwind) [reply web_src/less/_markup.less]: Git history points to it being related to webhook: https://github.com/go-gitea/gitea/commit/c2ca103d30b0f47b5ad0fa544ab5a07263894ea3.

Comment (silverwind) [reply web_src/less/_markup.less]: Webhook delivery list looks fine with the removal, let's do this.

Comment (silverwind) [reply web_src/less/_markup.less]: Pushed this change.

**Comment by silverwind** [comment]
More changes pushed:

- Rename `markdown` to `markup` everywhere
- Simplify docs by not suggesting to load Less code in the browser, that's just horrible advice.

File: templates/repo/view_file.tmpl [hunk 3]
```diff
@@ -64,7 +64,7 @@
 		{{end}}
 	</h4>
 	<div class="ui attached table unstackable segment">
-		<div class="file-view {{if .IsMarkup}}markup {{.MarkupType}} {{if ne "csv" .MarkupType}}{{end}}{{else if .IsRenderedHTML}}plain-text{{else if .IsTextSource}}code-view{{end}}">
```
Comment (silverwind) [inline templates/repo/view_file.tmpl]: `{{if ne "csv" .MarkupType}}{{end}}` was noop here, fyi.

**Review by silverwind — APPROVED** [review]

File: web_src/less/markup/content.less [hunk 4]
```diff
@@ -477,12 +477,37 @@
...
+
+  .csv-data tr {
+    border-top: 0;
+  }
+
+  .csv-data th {
+    font-weight: 600;
+    border-top: 0;
+  }
+
```
Comment (silverwind) [inline web_src/less/markup/content.less]: Where do these come from? I see no related CSS class changes, are they dead code?

Comment (6543) [reply web_src/less/markup/content.less]: no idear either

Comment (silverwind) [reply web_src/less/markup/content.less]: I think @HarvsG might have added them during testing and Github's CSS includes a same-name class but given that we never actually output such HTML in first place, let's remove it. If  CSV rendering it added to our markdown, the CSS for it should be added along with that.

Comment (silverwind) [reply web_src/less/markup/content.less]: Removed them.

**Review by 6543 — COMMENTED** [review]
LGTM from my side ... :)

File: web_src/js/markup/anchors.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ function scrollToAnchor() {
 }
 
 export default function initMarkdownAnchors() {
```
Comment (lunny) [inline web_src/js/markup/anchors.js]: `initMarkupAnchors`?

**Comment by lunny** [comment]
For wiki and issue/comment, maybe we should still keep in `markdown`? Or both `markup markdown` because we currently only support markdown syntax in these places.

**Review by zeripath — APPROVED** [review]

File: web_src/js/markup/anchors.js (hunk 5 above)
Comment (silverwind) [reply web_src/js/markup/anchors.js]: Fixed, renamed and changed these to named exports.

**Comment by silverwind** [comment]
> For wiki and issue/comment, maybe we should still keep in markdown? Or both markup markdown because we currently only support markdown syntax in these places.

It would have to be both but there's probably not much of a benefit doing so. We use double classes for the cases where the user can define the format, which I think is only for repo (and lfs) content.

**Pull request merged**